Octokit.Tests.Integration/Clients/RepositoriesClientTests.cs

```diff
@@ -1371,6 +1371,93 @@ public async Task GetsEmptyLanguagesWhenNoneWithRepositoryId()
         }
     }
 
+    public class TheReplaceAllTopicsMethod : IDisposable
+    {
+        readonly IGitHubClient _github = Helper.GetAuthenticatedClient();
+        readonly RepositoryTopics _defaultTopics = new RepositoryTopics(new List<string> { "blog", "ruby", "jekyll" });
+        const string theRepoOwner = "SeanKilleen";
+        const string theRepository = "seankilleen.github.io";
+
+        [Fact]
+        public async Task ClearsTopicsWithAnEmptyList()
+        {
+            var result = await _github.Repository.ReplaceAllTopics(theRepoOwner, theRepository, new RepositoryTopics());
+            Assert.Empty(result.Names);
+
+            var doubleCheck = await _github.Repository.GetAllTopics(theRepoOwner, theRepository);
+            Assert.Empty((doubleCheck.Names));
+        }
+
+        [Fact]
+        public async Task ClearsTopicsWithAnEmptyListWhenUsingRepoId()
+        {
+            var repo = await _github.Repository.Get(theRepoOwner, theRepository);
+            var result = await _github.Repository.ReplaceAllTopics(repo.Id, new RepositoryTopics());
+            Assert.Empty(result.Names);
+
+            var doubleCheck = await _github.Repository.GetAllTopics(theRepoOwner, theRepository);
+            Assert.Empty((doubleCheck.Names));
+        }
+
+        [Fact]
+        public async Task ReplacesTopicsWithAList()
+        {
+            var defaultTopicsList = new RepositoryTopics(_defaultTopics.Names);
+            var result = await _github.Repository.ReplaceAllTopics(theRepoOwner, theRepository, defaultTopicsList);
+
+            Assert.NotEmpty(result.Names);
+            Assert.Contains(result.Names, item => _defaultTopics.Names.Contains(item, StringComparer.InvariantCultureIgnoreCase));
+
+            var doubleCheck = await _github.Repository.GetAllTopics(theRepoOwner, theRepository);
```

Comment:
❓ How do we feel about the double-checks like this one here? Keep? Move to a separate test? Remove altogether?

Reply:
Seems legit to me. It's testing that the method actually persisted the changes and didn't just return the replaced topics. :smile:

Reply:
I think this is fine